app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -33,6 +33,10 @@ def provider_object(connection = nil)
     (connection || connection_source.connect).api.credentials.find(manager_ref)
   end
 
+  def manager_ref
+    self[:manager_ref].to_i
```

Comment:
We have tested with this change in a few scenarios and it resolves the reported issue.  👍 

@jameswnl Is there any condition you can think of where this column could end up as `nil` and we may not want to convert that to an int?

Reply:
`nil.to_i` returns 0 and also this is implemented in `Tower` subclass.

But talked to @agrare has concern on changing the type returned.  So your team may have to explore performing the conversion in automate code

Reply:
We would have to find all the places where we do something like: https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/automation_manager/playbook.rb#L49  (There are other places that do something similar.)

Maybe the other third option is to do this at the client gem layer so the database and code can continue the same but the client knows what Tower is expecting.

Reply:
@gmcculloug going Tower client gem route, you mean updating https://github.com/ansible/ansible_tower_client_ruby/blob/master/lib/ansible_tower_client/base_models/job_template.rb#L12  to search/replace the `options` param? 
hmm.. I am not sure if this is a clean way ...